File: arrays/prob1266.py
# 1266 Minimum time visiting all points

class Solution(object):
    def minTimeToVisitAllPoints(self, points):
        """
        :type points: List[List[int]]
        :rtype: int
        """
        m = len(points)
        if m == 1:
            return 0
        time = 0
        for i in range(m-1):
            dx = abs(points[i+1][0] - points[i][0])
            dy = abs(points[i+1][1] - points[i][1])
            time += max(dx, dy)
        return time

File: math/prob1266.kt
// 1266. Minimum Time Visiting All Points
import kotlin.math.abs

class Solution {
    fun minTimeToVisitAllPoints(points: Array<IntArray>): Int {

        val n = points.size
        var secs = 0

        for (i in 1 until n) {
            var dx = abs(points[i][0] - points[i-1][0])
            var dy = abs(points[i][1] - points[i-1][1])
            secs += if (dx >= dy) dx else dy
        }

        return secs


    }
}
